src/Relations/BelongsToThrough.php

```diff
@@ -8,8 +8,16 @@
 use Illuminate\Database\Eloquent\Relations\Concerns\SupportsDefaultModels;
 use Illuminate\Database\Eloquent\Relations\Relation;
 use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Database\Query\Expression;
 use Illuminate\Support\Str;
 
+/**
+ * @template TRelatedModel of \Illuminate\Database\Eloquent\Model
+ * @template TIntermediateModel of \Illuminate\Database\Eloquent\Model
```

Comment:
This isn't going to work when you have multiple intermediate models, I wonder if this could be an array:

```
 * @template TIntermediateModels of list<\Illuminate\Database\Eloquent\Model>
 ```

Then to use:
```
@return \Znck\Eloquent\Relations\BelongsToThrough<..., array{0: \Tests\Models\VendorCustomerAddress}, ...>
```